src/Loop.php

```diff
@@ -59,50 +35,14 @@ public static function setFactory(DriverFactory $factory = null)
      * exception is thrown that cannot be handled. Exceptions that cannot be handled are exceptions thrown from an
      * error handler or exceptions that would be passed to an error handler but none exists to handle them.
      *
-     * @param callable $callback The callback to execute.
-     * @param Driver $driver The event loop driver. If `null`, a new one is created from the set factory.
-     *
      * @return void
      *
      * @see \AsyncInterop\Loop::setFactory()
      */
-    public static function execute(callable $callback, Driver $driver = null)
-    {
-        $previousDriver = self::$driver;
-
-        self::$driver = $driver ?: self::createDriver();
-        self::$level++;
-
-        try {
-            self::$driver->defer($callback);
-            self::$driver->run();
-        } finally {
-            self::$driver = $previousDriver;
-            self::$level--;
-        }
-    }
-
-    /**
-     * Create a new driver if a factory is present, otherwise throw.
-     *
-     * @return Driver
-     *
-     * @throws \Exception If no factory is set or no driver returned from factory.
-     */
-    private static function createDriver()
+    public static function run()
     {
-        if (self::$factory === null) {
-            throw new \Exception("No loop driver factory set; Either pass a driver to Loop::execute or set a factory.");
-        }
-
-        $driver = self::$factory->create();
-
-        if (!$driver instanceof Driver) {
-            $type = is_object($driver) ? "an instance of " . get_class($driver) : gettype($driver);
-            throw new \Exception("Loop driver factory returned {$type}, but must return an instance of Driver.");
-        }
-
-        return $driver;
+        $driver = self::$driver ?: self::get();
```

Comment:
What is the purpose of this `get()` call if it just returns the value of `$driver`?